ToolBox/QuiverDataDownloader/QuiverWallStreetBetsDataDownloader.cs

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using QuantConnect.Configuration;
+using QuantConnect.Data.Auxiliary;
+using QuantConnect.Data.Custom.Quiver;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QuantConnect.ToolBox.QuiverDataDownloader
+{
+
+    public class QuiverWallStreetBetsDataDownloader : QuiverDataDownloader
+    {
+        private readonly string _destinationFolder;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="QuiverWikipediaDataDownloader"/>
+        /// </summary>
+        /// <param name="destinationFolder">The folder where the data will be saved</param>
+        public QuiverWallStreetBetsDataDownloader(string destinationFolder)
+        {
+            _destinationFolder = Path.Combine(destinationFolder, "wallstreetbets");
+
+            Directory.CreateDirectory(_destinationFolder);
+        }
+
+        /// <summary>
+        /// Runs the instance of the object.
+        /// </summary>
+        /// <returns>True if process all downloads successfully</returns>
+        public override bool Run()
+        {
+            var stopwatch = Stopwatch.StartNew();
+
+            try
+            {
+                var companies = GetCompanies().Result.DistinctBy(x => x.Ticker).ToList();
+                var count = companies.Count;
+                var currentPercent = 0.05;
+                var percent = 0.05;
+                var i = 0;
+
+                Log.Trace($"QuiverWallStreetBetsDataDownloader.Run(): Start processing {count.ToStringInvariant()} companies");
+
+                var tasks = new List<Task>();
+
+                foreach (var company in companies)
+                {
+                    // Include tickers that are "defunct".
+                    // Remove the tag because it cannot be part of the API endpoint.
+                    // This is separate from the NormalizeTicker(...) method since
+                    // we don't convert tickers with `-`s into the format we can successfully
+                    // index mapfiles with.
+                    var quiverTicker = company.Ticker;
+                    string ticker;
+
+
+                    if (!TryNormalizeDefunctTicker(quiverTicker, out ticker))
+                    {
+                        Log.Error($"QuiverWallStreetBetsDataDownloader(): Defunct ticker {quiverTicker} is unable to be parsed. Continuing...");
+                        continue;
+                    }
+
+                    // Begin processing ticker with a normalized value
+                    Log.Trace($"QuiverWallStreetBetsDataDownloader.Run(): Processing {ticker}");
+
+                    // Makes sure we don't overrun Quiver rate limits accidentally
+                    System.Threading.Thread.Sleep(1);
```

Comment:
```
 // Makes sure we don't overrun Quiver rate limits accidentally
System.Threading.Thread.Sleep(1);
```
Shouldn't be required since we are using the `IndexGate.WaitToProceed();`